### PR TITLE
Merge Development to Master

### PR DESCRIPTION
**Pull request by peterOutOfLowerGumtree**

### CODE REVIEW

File: src/main/scala/Cipher.scala [hunk 1]
```diff
@@ -0,0 +1,17 @@
+
+class Cipher {
+
+  val cipherCode: Map[Char, Char] = (('a' to 'z').toList zip ('a' to 'z').reverse).toMap
```
Comment (Imgundams) [inline src/main/scala/Cipher.scala]: This is a nice way to get the map :+1:

File: src/test/scala/CipherPosTests.scala [hunk 2]
```diff
@@ -0,0 +1,20 @@
+
+import org.scalatest.{FlatSpec, Matchers}
+
+class CipherPosTests extends FlatSpec with Matchers {
+
+  val cipher = new Cipher
+
+  "hello" should "output the correct cipher" in {
```
Comment (Imgundams) [inline src/test/scala/CipherPosTests.scala]: The test names should have an clearly identifiable name, I have no idea what the hello test should do other then output the correct string.

File: src/test/scala/CipherPosTests.scala [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    cipher.convert("hello") shouldBe "svool"
+  }
+
+  "HELLO" should "output the correct cipher" in {
+    cipher.convert("HELLO") shouldBe "svool"
+  }
+
+  "James's shoes" should "output the correct cipher" in {
```
Comment (Imgundams) [inline src/test/scala/CipherPosTests.scala]: You tried :+1:

File: src/test/scala/NumberFailTests.scala [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+  "1234e" should "output error message" in {
+    num.checkIfNumber("1234e", num.french) shouldBe "Enter a valid positive number"
+  }
+
+  "5747836578364529623586345786342562356029345" should "output error message" in {
+    num.checkIfNumber("5747836578364529623586345786342562356029345", num.english) shouldBe "Enter a valid positive number"
+  }
+
+  "" should "output nothing" in {
```
Comment (Imgundams) [inline src/test/scala/NumberFailTests.scala]: Odd name for a test

File: src/main/scala/NumberToLongShort.scala [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  def langOutput(input: String, lang: List[String]): String = {
+    if (input.contains('-')) {
+      langNeg(input, lang)
+    }
+    else {
+      langPos(input, lang)
+    }
+  }
+
+  def langPos(input: String, lang: List[String]): String = {
```
Comment (Imgundams) [inline src/main/scala/NumberToLongShort.scala]: `langPos` and `langNeg` look similar, could be done in one line sacrificing clarity for each `def`

File: src/test/scala/NumberPosTests.scala [hunk 6]
```diff
@@ -0,0 +1,54 @@
+
+import org.scalatest.{FlatSpec, Matchers}
+
+class NumberPosTests extends FlatSpec with Matchers {
+
+  val num = new NumberToLongShort
+
+  "8761234563486534711" should "output correct english result" in {
```
Comment (Imgundams) [inline src/test/scala/NumberPosTests.scala]: Test names for all these tests should be like `"For an Input of 8761234563486534711" should " output the correct result in English" in {`

File: src/main/scala/NumberToLongShort.scala [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+  def checkIfNumber(input: String, lang: List[String]): String = {
+    Try {
+      input.toLong
+    } match {
+      case Success(a) => langOutput(input, lang)
+      case Failure(a) => "Enter a valid positive number"
+    }
+  }
+
+  def langOutput(input: String, lang: List[String]): String = {
```
Comment (Imgundams) [inline src/main/scala/NumberToLongShort.scala]: Clean Code™ says that variable names and function names should't be abbreviated.

**Review by Imgundams — APPROVED** [review]
Overall good, code is good,  however the test names are lacking. :+1: 
:^} )

File: src/main/scala/NumberToLongShort.scala [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+    Try {
+      input.toLong
+    } match {
+      case Success(a) => langOutput(input, language)
+      case Failure(a) => "Enter a valid positive number"
+    }
+  }
+
+  def langOutput(input: String, language: List[String]): String = {
+    if (input.contains('-')) {
```
Comment (tvaidotas) [inline src/main/scala/NumberToLongShort.scala]: this would look nicer in pattern maching

File: src/main/scala/NumberToLongShort.scala [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+    }
+    else {
+      langPos(input, language)
+    }
+  }
+
+  def langPos(input: String, language: List[String]): String = {
+    val in = input.replaceFirst("^0*", "").reverse.grouped(3).toList.map(_.reverse.toList.mkString)
+    val out = calculation(in, language)
+    out.reverse.mkString
```
Comment (tvaidotas) [inline src/main/scala/NumberToLongShort.scala]: val out = calculation(in, language)
out.reverse.mkString
could just be
calculation(in, language).reverse.mkString
remember that the last statement is returned

File: src/main/scala/NumberToLongShort.scala [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+  }
+
+  def langNeg(input: String, language: List[String]): String = {
+    val in = input.tail.replaceFirst("^0*", "").reverse.grouped(3).toList.map(_.reverse.toList.mkString)
+    val out = calculation(in, language)
```
Comment (tvaidotas) [inline src/main/scala/NumberToLongShort.scala]: similar to the previous comment

**Comment by tvaidotas** [comment]
looks good

**Pull request merged**